sys/cheri/cheric.h

```diff
@@ -34,6 +34,9 @@
 
 #include <sys/cdefs.h>
 #include <sys/types.h>
+#if !defined(_KERNEL) && !defined(_STANDALONE)
```

Comment:
Should I add a !defined(__cplusplus) here? I don't think it really matters since it will just be a no-op for C++.